c7n/resources/dms.py

```diff
@@ -105,3 +121,114 @@ def process(self, resources):
         client = local_session(self.manager.session_factory).client('dms')
         for arn, r in zip(self.manager.get_arns(resources), resources):
             client.delete_replication_instance(ReplicationInstanceArn=arn)
+
+
+@ReplicationInstance.action_registry.register('tag')
+class InstanceTag(Tag):
+    """
+    Add tag(s) to a replication instance
+
+    :example:
+
+        .. code-block:: yaml
+
+            policies:
+                - name: tag-dms-required
+                  resource: dms-instance
+                  filters:
+                    - "tag:RequireTag": absent
+                  actions:
+                    - type: tag
+                      key: RequiredTag
+                      value: RequiredTagValue
+    """
+    permissions = ('dms:AddTagsToResource',)
+
+    def process_resource_set(self, resources, tags):
+        client = local_session(self.manager.session_factory).client('dms')
+        tags_list = []
+        for t in tags:
+            tags_list.append({'Key': t['Key'], 'Value': t['Value']})
+        for r in resources:
+            try:
+                client.add_tags_to_resource(
+                    ResourceArn=r['ReplicationInstanceArn'],
+                    Tags=tags_list)
+            except Exception as e:
```

Comment:
can you use the ClientError instead of a naked exception, it prevents masking other errors than the expected ones.

`from botocore.exceptions import ClientError`

and then 

`except ClientError as e:`

Reply:
Yeah, I noticed I was doing that a lot in the other PRs as well. Will correct shortly.